### PR TITLE
Improved consistency and fixed some equations

### PR DESCRIPTION
**Pull request by mglasser16**

### CODE REVIEW

**Comment by mglasser16** [comment]
https://drive.google.com/file/d/1D33JzrY7bFIHW2SNpRIseOMJRcdaICtc/view?usp=sharing
List of equations

File: Nucleation_model.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import random as ran
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+from scipy.integrate import solve_ivp
+#%%
+"""=========================== Constants ==========================="""
+
+k_B = 1.380649E-23 #J K-1 // Boltzmann's constant
+ep_0 = 8.854E-12  #F m-1 // electric permittivity of a vacuum
+pi = 3.141529
```
Comment (decaluwe) [inline Nucleation_model.py]: the `math` package includes pi, so you can just use `m.pi` for greater precision.

File: Nucleation_model.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+N_a = 6.022E23 #mol-1 // Avogadro's number
+R = 8.3145 #J mol-1 K-1 // Ideal constant
+
+"""=========================== User system variables ==========================="""
```
Comment (decaluwe) [inline Nucleation_model.py]: I call out a few specific varaibles, below, but in general, I would like comments to give slightly more detailed information about specifically what each input variable represents.

File: Nucleation_model.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+#Parameters that can be changed during the trials
+T = 25 + 273.15 #K
+C_LiO2 = 0.15 #mol m-3
+C_Li = 0.15 #mol m-3
+Elyte_v = 0.0005 #m3
+N_0 = 0
```
Comment (decaluwe) [inline Nucleation_model.py]: What does this represent?

File: Nucleation_model.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+pi = 3.141529
+N_a = 6.022E23 #mol-1 // Avogadro's number
+R = 8.3145 #J mol-1 K-1 // Ideal constant
+
+"""=========================== User system variables ==========================="""
+#Parameters that can be changed during the trials
+T = 25 + 273.15 #K
+C_LiO2 = 0.15 #mol m-3
+C_Li = 0.15 #mol m-3
+Elyte_v = 0.0005 #m3
```
Comment (decaluwe) [inline Nucleation_model.py]: Probably better to let a user input this in units that make sense to them (mL?  uL?) and then convert later.

File: Nucleation_model.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+N_0 = 0
+A_0 = pi*0.009**2 #m2
```
Comment (decaluwe) [inline Nucleation_model.py]: Similar to above, probably better to let the user input an intial radius?

File: Nucleation_model.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+time = 100000
+
+"""=========================== User thermodyanic imputs ==========================="""
+#parameters that depend on the system being studied
+
```
Comment (decaluwe) [inline Nucleation_model.py:34]: Again, for the section below, give the user detailed info on each input variable, so they know they are giving the correct desired inputs.

File: Nucleation_model.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+C_Li_sat = 0.1
+k_nu = 1E-6 #mol s-1 m-2 // Yin (2017)
+gamma_surf = 7.7E-3 #J m-2 // Danner (2019) - should be replaced for LiS system
+V = 1.98E-5 #m3 mol-1 // Yin (2017)
+theta = 30*pi/180 # radians // Danner (2019) - should be replaced for LiS system
+D_LiO2 = 1.2E-9 #m2 -s-1 // Yin (2017)
+
+
+"""=========================== Thermodyanic system cals ==========================="""
+
```
Comment (decaluwe) [inline Nucleation_model.py:50]: At the risk of sounding like a broken record... 😜

File: Nucleation_model.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+N_sites_0 =  A_0/(pi*r_crit**2)
+
+"""=========================== Initialization step ==========================="""
+#%%
+#intializing the solution vector
+initial = {}
+initial['N'] = N_0
+initial['N_sites'] = N_sites_0
+sol_vec = list(initial.values())  # solution vector
```
Comment (decaluwe) [inline Nucleation_model.py:60]: Interesting method!  Is there a way for us to know the order in which the variables are stored, here?

File: Nucleation_model.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+print(sol_vec)
+
+"""=========================== Equations ==========================="""
+#%%
+
+
+def residual(t, solution):
+    N, N_sites = solution
+    DN_Dt = D_LiO2*a_d*N_sites*Z*m.exp(-Del_G_Crit/(k_B*T))
+    DN_sites_Dt = - DN_Dt
```
Comment (decaluwe) [inline Nucleation_model.py]: Do we really need to store this? It seems like it is not independent of the `N` variable.

That said, maybe when we factor in growth, the two will become decoupled?

File: Nucleation_model.py [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    return [DN_Dt, DN_sites_Dt]
+
+solution = solve_ivp(residual, [0, time], sol_vec)
+
+nucleations = solution.y[0]
```
Comment (decaluwe) [inline Nucleation_model.py:88]: Just a note: eventually we'd probably like some type of 'pointer' structure that tells us, in a generalized way, where in the solution each variable is stored.  Not important right now, but eventually...

**Review by decaluwe — CHANGES_REQUESTED** [review]
Excellent work, @mglasser16 !

I have a fair number of comments, below, but really this is in very good shape.  Improve the comments explaining the various variables, and then I think this is ready to merge.

👏👏👏

**Comment by decaluwe** [comment]
So I think the next thing I would recommend is adding governing equations for the electrolyte concentration.  Demonstrate that after a certain period of nucleation, the ion concentration in the electrolyte drops to a point where nucleation shuts off.

File: Nucleation_model.py (hunk 5 above)
Comment (mglasser16) [reply Nucleation_model.py]: I initially did this but wasn't sure if I could assume that the user would be working with a circular electrode

File: Nucleation_model.py (hunk 8 above)
Comment (mglasser16) [reply Nucleation_model.py:60]: Yes, I will be updating this to be DA_dt, and recalculating R_crit in the future as R_crit is related to C_LiO2/C_sat!

**Comment by decaluwe** [comment]
This looks great.  See the message from my commit to your repo, but in short, you were assigning the particle radius to `R`, but then also trying to use `R` as the Universal Gas Constant, which resulted in division by zero.

Fixed it by using `Rbar` for the gas constant.

Also changed the solver used (default is RK45, switched it to BDF, which worked better for your problem).

Nice work!

**Pull request merged**